mw-webapp/src/pages/authPage/AuthPage.tsx

```diff
@@ -1,17 +1,37 @@
-import {Link} from "react-router-dom";
-import {useAuth} from "src/utils/useAuth";
+import {User} from "firebase/auth";
+import {useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
+import {Button} from "src/component/button/Button";
+import {getNewUserCredentials} from "src/utils/auth/getNewUserCredentials";
+import {handleUserAuthState} from "src/utils/auth/handleUserAuthState";
+import {handleLogIn} from "src/utils/auth/handleLogIn";
+import {handleLogOut} from "src/utils/auth/handleLogOut";
 
 export const AuthPage = () => {
-  const handleGoogleSignIn = useAuth();
+  const [user, setUser] = useState<User | null>(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    handleUserAuthState(setUser);
+    return () => {
+      getNewUserCredentials();
+    };
```

Comment:
Do you need to use return getNewUserCredentials or maybe just call it will be enough

Reply:
Now it's enough. Early this function doesn't write user to DB if I call it in useEffect.
Done